### PR TITLE
feat(profiling): continuous profile serialization schema

### PR DESCRIPTION
**Pull request by armcknight**
#skip-changelog, for #3555

Implements the necessary schema changes for how continuous profiling chunks will be serialized, and their envelope structure. This was cribbed from the current functions that do the same for legacy profiles (`sentry_serializedProfileDataLegacy` and `sentry_profileEnvelopeItemLegacy`) to create `sentry_serializedContinuousProfileChunk` and `sentry_continuousProfileChunkEnvelope`.

Also implements the timer logic to transmit those envelopes every time it fires in `SentryProfiler.mm`.

Added assertions for the new schema to `SentryContinuousProfilerTests.swift`. Those tests will simulate sending multiple chunks per test, validating the contents of each, which have unique data per chunk to ensure the same data isn't carrying over, and that the profiler is still running after transmitting each until the profiler is explicitly stopped.

### CODE REVIEW

File: Sources/Sentry/Profiling/SentryProfilerSerialization.mm
Comment (philipphofmann) [inline Sources/Sentry/Profiling/SentryProfilerSerialization.mm:240]: `m`: We could significantly reduce the payload if we only add the debug images references in the samples payload; see `SentryDebugImageProvider.getDebugImagesForFrames`, but we don't have to do this in this PR.

File: Sources/Sentry/Profiling/SentryProfilerSerialization.mm
Comment (philipphofmann) [inline Sources/Sentry/Profiling/SentryProfilerSerialization.mm:249]: `m`: Shouldn't we also add os and device data https://github.com/getsentry/sentry-cocoa/blob/4c0b15137001db76ed776f24dd430b9121a977ba/Sources/Sentry/Profiling/SentryProfilerSerialization.mm#L139-L152

If yes, I think we should extract this into a method, to reduce duplication.

File: Sources/Sentry/Profiling/SentryProfilerSerialization.mm
Comment (philipphofmann) [inline Sources/Sentry/Profiling/SentryProfilerSerialization.mm:252]: `m`: The metrics code is a bit different from `sentry_serializedProfileDataLegacy`. Is this intended? If yes, why? If not, I think we should extract some method to get rid of some duplication instead of duplicating it.

File: Sources/Sentry/Profiling/SentryProfilerSerialization.mm
Comment (philipphofmann) [inline Sources/Sentry/Profiling/SentryProfilerSerialization.mm:259]: `m`: Codecov complains quite a bit for uncovered lines in this method. Should we add tests for them?

File: Sources/Sentry/SentryProfiler.mm
Comment (philipphofmann) [inline Sources/Sentry/SentryProfiler.mm:261]: `m`: Does it make sense to move this to a background thread to potentially unblock the calling thread, which could be the main thread to tackle https://github.com/getsentry/sentry-cocoa/issues/3893? Also, okay to do this in another PR.

File: Sources/Sentry/include/HybridPublic/SentryEnvelopeItemType.h
Comment (philipphofmann) [inline Sources/Sentry/include/HybridPublic/SentryEnvelopeItemType.h:10]: `h`: We must add a data category related to this envelope item type or map this envelope item type to the data category profile so rate limiting and client reports work correctly. See https://github.com/getsentry/sentry-cocoa/blob/e072ad1b27ff5e6413edd2e4716a5ef0be4eee04/Sources/Sentry/SentryDataCategoryMapper.m#L35-L37

File: Tests/SentryProfilerTests/SentryContinuousProfilerTests.swift
Comment (philipphofmann) [inline Tests/SentryProfilerTests/SentryContinuousProfilerTests.swift:84]: `l`: Asserting for the expected version depending on continuous profiling or tracer based profiling would improve the tests.

**Review by philipphofmann — COMMENTED** [review]
A couple of issues, but looking good already 👍

File: Sources/Sentry/Profiling/SentryProfilerSerialization.mm
Comment (armcknight) [reply Sources/Sentry/Profiling/SentryProfilerSerialization.mm:106]: I did rename it there, but it's stacked on top of this pr. it'll have to be either merged after this one, or combined into it (which I'd prefer not to do)

File: Sources/Sentry/Profiling/SentryProfilerDefines.h
Comment (armcknight) [reply Sources/Sentry/Profiling/SentryProfilerDefines.h:9]: See https://github.com/getsentry/sentry-cocoa/pull/3932#discussion_r1597527928

File: Sources/Sentry/Profiling/SentryProfilerSerialization.mm
Comment (armcknight) [reply Sources/Sentry/Profiling/SentryProfilerSerialization.mm:240]: This is good feedback. Maybe we send the full list of debug images in the first chunk, and then in subsequent chunks, send a list of images that have been added or removed since the last chunk. @phacops what do you think?

In any case I think this will need to be done in a subsequent PR, and any change is going to depend on a change in the backend as well, so we shouldn't hold up this PR for it. I've made a note in the [planning doc](https://www.notion.so/sentry/Continuous-Profiling-Cocoa-SDK-9d64a809ed2d487b89e4aa0358bfac18?pvs=4#c2725f3879c442729e8bd107dd56ea8c).

File: Sources/Sentry/Profiling/SentryProfilerSerialization.mm
Comment (armcknight) [reply Sources/Sentry/Profiling/SentryProfilerSerialization.mm]: we are no longer sending those in profile payloads, they're going to get it elsewhere in the backend.

File: Sources/Sentry/Profiling/SentryProfilerSerialization.mm
Comment (armcknight) [reply Sources/Sentry/Profiling/SentryProfilerSerialization.mm:252]: Here are the differences:
- instead of using dictionary literals, which have led to crashes in the past, I went with NSMutableDictionaries and added the same fields with assignment
- the units are in milliseconds instead of nanoseconds (this change applies to all of continuous profiling durations)

File: Sources/Sentry/SentryProfiler.mm
Comment (armcknight) [reply Sources/Sentry/SentryProfiler.mm:261]: Yes, I'm definitely on board to do that, but agree it is appropriate to do so in another PR. ([link](https://www.notion.so/sentry/Continuous-Profiling-Cocoa-SDK-9d64a809ed2d487b89e4aa0358bfac18?pvs=4#5454422467754884b59f496ddaa77b0b))

File: Sources/Sentry/include/HybridPublic/SentryEnvelopeItemType.h
Comment (philipphofmann) [inline Sources/Sentry/include/HybridPublic/SentryEnvelopeItemType.h:1]: Thanks for the comment

File: Sources/Sentry/SentryDataCategoryMapper.m
Comment (philipphofmann) [inline Sources/Sentry/SentryDataCategoryMapper.m:14]: Didn't know that data category exists already 💯 
https://github.com/getsentry/relay/blob/23a48f2c7142ab940d5e6e3d3289c087eba2fcf2/py/sentry_relay/consts.py#L30

**Review by philipphofmann — APPROVED** [review]
LGTM, thanks 👍

File: Sources/Sentry/Profiling/SentryProfilerSerialization.mm
Comment (armcknight) [reply Sources/Sentry/Profiling/SentryProfilerSerialization.mm:240]: Oh yes definitely after another app launch. I was only thinking about doing it between chunk envelopes in the same profile session (another app launch implies a different profile session)

File: Sources/Sentry/Profiling/SentryProfilerSerialization.mm
Comment (armcknight) [reply Sources/Sentry/Profiling/SentryProfilerSerialization.mm:259]: I'm beefing up the tests around this.

Comment (armcknight) [reply Sources/Sentry/Profiling/SentryProfilerSerialization.mm:259]: I have another branch at the top of the stack where i was already adding more tests, I will do these there because these particular tests involve a lot of work around timestamps, and we just made a decision that's going to affect that (moving to using NSDate for all profiling timestamps), which I'd like to merge before writing the tests to avoid potentially doing a lot of annoying debugging twice.

**Pull request merged**
